src/cgutils.cpp

```diff
@@ -1564,7 +1564,7 @@ static jl_cgval_t typed_store(jl_codectx_t &ctx,
             ret = mark_julia_type(ctx, callval, true, jl_any_type);
         }
         if (!jl_subtype(ret.typ, jltype)) {
-            emit_typecheck(ctx, ret, jltype, fname + "typed_store");
+            emit_typecheck(ctx, ret, jltype, fname);
```

Comment:
So this fixes the error message

```julia
julia> mutable struct AtomicRef{T}
           @atomic value::T
       end

julia> function f!(r)
           + = (_, x) -> x
           @atomic r.value += :a
       end;

julia> f!(r)
ERROR: TypeError: in modifyfield!typed_store, expected Int64, got a value of type Symbol
```

to `TypeError: in modifyfield!, expected Int64...`? Sounds like an improvement to me!